addons/mail/static/src/discuss/core/public/welcome_page.js

```diff
@@ -17,7 +17,7 @@ export class WelcomePage extends Component {
         this.rpc = useService("rpc");
         this.personaService = useService("mail.persona");
         this.state = useState({
-            userName: "Guest",
+            userName: this.props.data?.discussPublicViewData?.guest_name || "Guest",
```

Comment:
I probably won't ever be displayed now, but "Guest" here should be translatable.